did-key/src/lib.rs

```diff
@@ -98,6 +128,28 @@ impl DIDResolver for DIDKey {
                 x509_thumbprint_sha1: None,
                 x509_thumbprint_sha256: None,
             }
+        } else if data[0] == DID_KEY_SECP256K1_PREFIX[0] && data[1] == DID_KEY_SECP256K1_PREFIX[1] {
+            if data.len() - 2 != 33 {
```

Comment:
are we sure that this keylen holds true for all base58 reprs of secp256k1 pubkeys?

Reply:
e.g., i see "up to 35" in this analysis, but not a deterministic exact https://www.johndcook.com/blog/2019/03/04/base-58-encoding-and-bitcoin-addresses

Reply:
This line limits our implementation to accepting secp256k1 keys in compressed representation only (33 bytes). We could change it to also allow the uncompressed representation (65 bytes). We could also just remove this check since it is already checked when calling [secp256k1::PublicKey::parse_slice](https://docs.rs/libsecp256k1/0.3.5/secp256k1/struct.PublicKey.html#method.parse_slice), [here](https://github.com/spruceid/ssi/pull/93/files/bb250b27b1d8619e07fa07861539373adc700e39#diff-f2b45f70bcd251a75b708cd6f68019244b0b3430b7fd651ebab27ee5f3f814c9R31), which accepts an option for which representations to accept, using the [secp256k1::PublicKeyFormat](https://docs.rs/libsecp256k1/0.3.5/secp256k1/enum.PublicKeyFormat.html) enum (where currently the option is being passed to only accept the compressed format).

A secp256k1 public key has representations of different length, the compressed form and the uncompressed form. The `libsecp256k1` crate also supports a third [raw](https://docs.rs/libsecp256k1/0.3.5/secp256k1/enum.PublicKeyFormat.html#variant.Raw) representation which appears the uncompressed form but omitting the sign byte.

The (non-normative) [examples](https://w3c-ccg.github.io/did-method-key/#secp256k1) in `did:key` are using the compressed form. I looked into [did-key.js](https://github.com/transmute-industries/did-key.js/) as another implementation and it generates the compressed form too - although I didn't check if it it parses the uncompressed form also.
[did-key.rs](https://github.com/decentralized-identity/did-key.rs) actually produces the uncompressed form, but I might suggest that they change that. It also doesn't appear to parse either.